### DISCUSSION TITLE
std::ranges/concepts - keeping it modern

### DISCUSSION THREAD

**Discussion by itlods** [nlohmann/json#2623] [Ideas]
When dealing with the parsed data  both the the ranges::views and functional programming would assist in making terse readable code. I cannot see a fork or discussion that even mention these.
Concepts are a close cousin, in implementing this, and might be worth doing alone. Many issues are on type conversion and we could get much better errors from the compilers to assist programming.
Has anybody done some thoughts on if and how to add these?
Could we get away with some wrappers (for starters)?
How to stay c++11 compatible? Lots of #if or a clean fork?
...

**Comment by nlohmann** [comment]
So far, we added support for containers from C++17 with some `#ifdef`s, but think it will be difficult to have ranges in the same codebase, because this library targets C++11. But maybe I don't understand well enough what you want to achieve.
